deployment/cdk/opensearch-service-migration/lib/service-stacks/reindex-from-snapshot-stack.ts

```diff
@@ -70,17 +69,42 @@ export class ReindexFromSnapshotStack extends MigrationServiceCore {
         let rfsCommand = `/rfs-app/runJavaWithClasspath.sh com.rfs.RfsMigrateDocuments --s3-local-dir /tmp/s3_files --s3-repo-uri ${s3Uri} --s3-region ${this.region} --snapshot-name rfs-snapshot --lucene-dir '/lucene' --target-host ${osClusterEndpoint}`
         rfsCommand = parseAndMergeArgs(rfsCommand, props.extraArgs);
 
+        let targetUser = "";
+        let targetPassword = "";
+        let targetPasswordArn = "";
+        if (props.clusterAuthDetails.basic_auth) {
+            targetUser = props.clusterAuthDetails.basic_auth.username,
+            targetPassword = props.clusterAuthDetails.basic_auth.password? props.clusterAuthDetails.basic_auth.password : "",
+            targetPasswordArn = props.clusterAuthDetails.basic_auth.password_from_secret_arn? props.clusterAuthDetails.basic_auth.password_from_secret_arn : ""            
+        };
+
+        const openSearchPolicy = createOpenSearchIAMAccessPolicy(this.partition, this.region, this.account);
+        const openSearchServerlessPolicy = createOpenSearchServerlessIAMAccessPolicy(this.partition, this.region, this.account);
+        let servicePolicies = [artifactS3PublishPolicy, openSearchPolicy, openSearchServerlessPolicy];
+
+        const getSecretsPolicy = props.clusterAuthDetails.basic_auth?.password_from_secret_arn ? 
+            createTargetPasswordAccessPolicy(props.clusterAuthDetails.basic_auth.password_from_secret_arn) : null;
+        if (getSecretsPolicy) {
+            servicePolicies.push(getSecretsPolicy);
+        }
+
         this.createService({
             serviceName: 'reindex-from-snapshot',
             taskInstanceCount: 0,
             dockerDirectoryPath: join(__dirname, "../../../../../", "DocumentsFromSnapshotMigration/docker"),
-            dockerImageCommand: ['/bin/sh', '-c', rfsCommand],
+            dockerImageCommand: ['/bin/sh', '-c', "/rfs-app/entrypoint.sh"],
             securityGroups: securityGroups,
             taskRolePolicies: servicePolicies,
             cpuArchitecture: props.fargateCpuArch,
             taskCpuUnits: 1024,
             taskMemoryLimitMiB: 4096,
             ephemeralStorageGiB: 200,
+            environment: {
+                "RFS_COMMAND": rfsCommand,
```

Comment:
Are there any length constraints in the environment that we're introducing onto the command now?

Reply:
I'm not seeing any in the documentation so we may be fine here